conda/resolve.py

```diff
@@ -656,40 +675,48 @@ def generate_version_metrics(self, C, specs, include0=False):
                         rec.append(dist)
 
         for name, targets in iteritems(sdict):
-            pkgs = [(self.version_key(p), p) for p in self.groups.get(name, [])]
-            pkey = None
-            # keep in mind that pkgs is already sorted according to version_key (a tuple,
-            #    so composite sort key).  Later entries in the list are, by definition,
-            #    greater in some way, so simply comparing with != suffices.
-            for version_key, dist in pkgs:
-                if targets and any(dist == t for t in targets):
-                    continue
-                if pkey is None:
-                    ic = iv = ib = it = 0
-                # valid package, channel priority
-                elif pkey[0] != version_key[0] or pkey[1] != version_key[1]:
-                    ic += 1
-                    iv = ib = it = 0
-                # version
-                elif pkey[2] != version_key[2]:
-                    iv += 1
-                    ib = it = 0
-                # build number
-                elif pkey[3] != version_key[3]:
-                    ib += 1
-                    it = 0
-                elif pkey[4] != version_key[4]:
-                    it += 1
-
-                if ic or include0:
-                    eqc[dist.full_name] = ic
-                if iv or include0:
-                    eqv[dist.full_name] = iv
-                if ib or include0:
-                    eqb[dist.full_name] = ib
-                if it or include0:
-                    eqt[dist.full_name] = it
-                pkey = version_key
+            a_pkgs = [(self.version_key(p), p) for p in self.groups.get(name, [])]
+            f_pkgs = defaultdict(list)
+            if split_by_features:
```

Comment:
I think I'm confused by the design here. 

If `split_by_features` is True, then you create separate sort orders for each feature, and then when assigning the ic/iv/ib/it scores, you never use the features entry in the `VersionKey`. Since the `split_by_features` functionality is critical to your approach, it strikes me that you really didn't need to jump through all the hoops to _sort_ the packages differently.

So perhaps a simpler implementation---especially given that we're trying to preserve the original behavior---is to revert to the _original_ version key, _without_ the features inserted, and then just build your `f_pkgs` dictionary more directly.

Reply:
I wasn't sure whether the sort order of `Resolve.groups` is used or has influence on other parts of the code apart from `generate_version_metrics`. If that were the case, including the features before build number while sorting felt more natural for my intent to avoid comparing build number of different "feature-builds".
If you are very certain that the sort order does not influence other parts, we can remove the `VersonKey.features` value.